protofsm/msg_router.go

```diff
@@ -274,12 +274,20 @@ func (p *MultiMsgRouter) msgRouter() {
 			// to those that can handle it the message.
 			var couldSend bool
 			for _, endpoint := range endpoints {
-				if endpoint.CanHandle(msg) {
```

Comment:
Why does `CanHandle` need an error? The implementation should just be doing a type assertion on the message to decide if it can handle it or not.

Reply:
Yeah, the interface was changed to support returning an error, as this was required in tapd in order to allow for the server readiness check to fail
https://github.com/lightninglabs/taproot-assets/pull/1126/files#diff-366e46a40f6f60b4f7614eb0976bb51820364bf5ca6ccc4787eb49d7bdbef3e6R827-R845